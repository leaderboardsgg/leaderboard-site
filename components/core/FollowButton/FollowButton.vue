<template>
  <button
    class="follow-button"
    v-bind="$attrs"
    v-on="$listeners"
    @click="onClick"
    @keyup.enter="onClick"
  >
    Follow
  </button>
</template>

<script>
import { defineComponent } from '@nuxtjs/composition-api';

export default defineComponent({
  props: {
    onClick: {
      default: () => console.log('follow'), // eslint-disable-line no-console
      type: Function,
    },
  },
});
</script>

<style scoped>
.follow-button {
  @apply px-3 py-2 text-center border border-current rounded border-gray-300 rounded hover:bg-gray-100;
}
</style>
