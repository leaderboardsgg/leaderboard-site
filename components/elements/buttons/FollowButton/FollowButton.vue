<script setup lang="ts">
const emit = defineEmits<{
  (event: 'follow'): void
}>()
</script>

<template>
  <button
    class="follow-button"
    v-bind="$attrs"
    @click="emit('follow')"
    @keyup.enter="emit('follow')"
  >
    Follow
  </button>
</template>

<style lang="postcss" scoped>
.follow-button {
  @apply px-3 py-2 text-center border border-current rounded border-gray-300 hover:bg-gray-100;
}
</style>
