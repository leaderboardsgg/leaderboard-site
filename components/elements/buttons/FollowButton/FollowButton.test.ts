import { mountSuspended } from '@nuxt/test-utils/runtime'
import FollowButton from './FollowButton.vue'

describe('<FollowButton />', () => {
  it('should render without crashing', async () => {
    const wrapper = await mountSuspended(FollowButton, {
      props: {
        onClick: () => 69,
      },
    })

    expect(wrapper.isVisible()).toBe(true)
  })
})
