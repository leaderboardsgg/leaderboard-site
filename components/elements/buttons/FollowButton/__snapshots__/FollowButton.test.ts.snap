// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FollowButton /> renders correctly 1`] = `
<button
  class="follow-button"
>
  
  Follow

</button>
`;
